### DISCUSSION TITLE
Best place to store custom user data?

### DISCUSSION THREAD

**Discussion by dpotter-intoto** [netbox-community/netbox#11887] [Q&A] [answered]
We have a process that reviews a user's netbox change history and does some external validation.   I'd like to store a datetime with each user that indicates when his/her changes were last reviewed and validated, so that when the validation process runs, it can easily calculate what changes have been made since the last execution.

Although I could store this datetime in an external system, the ideal place would be data would be a custom field attached to the user object. But I don't think it's possible.

Any thoughts/recommendations?

Thanks!

**Comment by jeremystretch** [comment, marked as answer]
You're correct that it's not currently possible to add custom fields on the user model. (This is due to a limitation concerning the user model specifically.)

For any system or feature that introduces its own data, the best solution is almost always a custom plugin. This will allow you to introduce a new model dedicated to storing any and all data necessary, and allow for future development.
